web/src/routes/RepositoryDetails/Settings/RepoEventExpiry.tsx

```diff
@@ -1,19 +1,35 @@
-import {FormGroup, TextInput, ValidatedOptions} from '@patternfly/react-core';
+import {
+  FormGroup,
+  TextInput,
+  ValidatedOptions,
+  HelperText,
+  HelperTextItem,
+} from '@patternfly/react-core';
 import {NotificationEventConfig} from 'src/hooks/UseEvents';
-import {useState} from 'react';
+import {useEffect, useState} from 'react';
 
 export default function RepoEventExpiry(props: RepoEventExpiryProps) {
-  const [valid, setValid] = useState(ValidatedOptions.default);
+  const [valid, setValid] = useState<ValidatedOptions>(
+    ValidatedOptions.default,
+  );
 
-  const onChange = (_event, value) => {
-    props.setEventConfig({days: Number(value)});
+  const validateNumber = (value) => {
     if (value < 1) {
       setValid(ValidatedOptions.error);
     } else {
       setValid(ValidatedOptions.success);
     }
   };
 
+  const onChange = (_event, value) => {
+    props.setEventConfig({days: Number(value)});
+    validateNumber(value);
+  };
+
+  useEffect(() => {
```

Comment:
shouldn't the `useEffect` be run only when the text input for days change and not on every render?